tests/scripts/thread-cert/border_router/MATN_16_LargeNumberOfMulticastGroupSubscriptionsToBBR.py

```diff
@@ -0,0 +1,231 @@
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS 'AS IS'
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import logging
+import unittest
+
+import pktverify
+from pktverify import packet_verifier, packet_filter, consts
+from pktverify.consts import MAS, MA2
+import config
+import thread_cert
+
+# Test description:
+# The purpose of this test case is to verify that the Primary BBR can handle a
+# large number (75) of subscriptions to different multicast groups. Multicast
+# registrations are performed each time with 15 multicast addresses per
+# registration request.
+#
+# Topology:
+#
+#
+#    ----------------(eth)--------------------
+#           |                     |
+#         BR_1 (Leader)           |
+#           |                   Host
+#           |
+#           |
+#         Router_1
+#
+#
+#
+
+BR_1 = 1
+ROUTER_1 = 2
+HOST = 3
+
+REG_DELAY = 10
+
+
+class MATN_16_LargeNumberOfMulticastGroupSubscriptionsToBBR(thread_cert.TestCase):
+    USE_MESSAGE_FACTORY = False
+
+    TOPOLOGY = {
+        BR_1: {
+            'name': 'BR_1',
+            'is_otbr': True,
+            'allowlist': [ROUTER_1],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        ROUTER_1: {
+            'name': 'Router_1',
+            'allowlist': [BR_1],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        HOST: {
+            'name': 'Host',
+            'is_host': True,
+        },
+    }
+
+    def test(self):
+        br1 = self.nodes[BR_1]
+        router1 = self.nodes[ROUTER_1]
+        host = self.nodes[HOST]
+
+        br1.set_backbone_router(reg_delay=REG_DELAY, mlr_timeout=consts.MLR_TIMEOUT_MIN)
+        br1.start()
+        self.simulator.go(5)
+        self.assertEqual('leader', br1.get_state())
+        self.assertTrue(br1.is_primary_backbone_router)
+
+        router1.start()
+        self.simulator.go(5)
+        self.assertEqual('router', router1.get_state())
+
+        host.start(start_radvd=False)
+        self.simulator.go(5)
+
+        # Loop steps 1 to 4 with i := 1 to 5.
+        for i in range(1, 6):
+            # 1. Router_1 registers for 15 multicast addresses, MASi.
+            router1.register_multicast_listener(*(MAS[i][1:]), timeout=consts.MLR_TIMEOUT_MIN)
```

Comment:
```suggestion
            self.assertEqual((0, []), router1.register_multicast_listener(*(MAS[i][1:]), timeout=consts.MLR_TIMEOUT_MIN))
```
make sure it's successful